Automattic/wp-calypso
# Happychat: Add unread message notification

Adds a visual indication when there are unread notifications in Happychat. This PR is another approach to the work done months ago in https://github.com/Automattic/wp-calypso/pull/10903.

This approach defines a "focused" and "blurred" state for Happychat. Any messages that come in while HC is "blurred" are considered unread. Currently we only set these states when HC opens or closes — but we may consider also setting/removing focus in other circumstances, e.g. when the browser window loses and gains focus.

GIF example:
![hc unread notifications](https://cloud.githubusercontent.com/assets/518059/24987071/69e9572a-1fc3-11e7-99c6-95b1dd884b4f.gif)


### To test

#### Test the primary HC interface
- Go to http://calypso.localhost:3000/help/contact and start a new Happychat session
- Respond from the Operator — the notification icon should not appear while Happychat is visible
- Minimize Happychat
- Respond from the Operator — the notification icon should appear
- Click the sidebar footer chat icon — this should open HC and clear the notification
- Minimize Happychat again
- You should be on `/help` page now — click "Contact Us" button (you need to get back to a page that will re-connect Happychat until https://github.com/Automattic/wp-calypso/pull/13020 merges)
- Respond again from the Operator — the notification icon should appear
- Refresh the Calypso page — after the refresh you should still see the notification icon
- Click the sidebar footer chat icon — this should open HC and clear the notification

#### Test the standalone HC page
- Go to http://calypso.localhost:3000/me/chat and start a new Happychat session
- During this session, the unread notification should not appear
- Navigate away to another page by clicking a sidebar link
- Respond from the Operator — the notification icon should appear
